### PR TITLE
Add cloud volume feature

### PR DESCRIPTION
**Pull request by jdenver23**
Depends on:
- [x] https://github.com/ManageIQ/manageiq/pull/21923

### CODE REVIEW

**Comment by agrare** [comment]
Hey @jdenver23 it looks like you merged master into this branch rather than rebase this branch onto master.  Recommend using `git config --global --add pull.rebase true` so that `git pull upstream master` will rebase your branch onto master without any extra merge commits.

File: app/models/manageiq/providers/ibm_cloud/power_virtual_servers/storage_manager/cloud_volume.rb
Comment (jaywcarman) [inline app/models/manageiq/providers/ibm_cloud/power_virtual_servers/storage_manager/cloud_volume.rb:116]: I would prefer to call this `_('Volume Base Name')` and then add some "helperText" so the user can determine what the cloned volume name will be.
```ruby
:helperText => _("Base name of the new cloned volume(s). Cloned Volume names will be prefixed with 'clone-' and suffixed with '-#####' (where ##### is a 5 digit random number)")
```

File: app/models/manageiq/providers/ibm_cloud/power_virtual_servers/storage_manager/cloud_volume.rb
Comment (jaywcarman) [inline app/models/manageiq/providers/ibm_cloud/power_virtual_servers/storage_manager/cloud_volume.rb:195]: Either need to define a `MiqException::MiqVolumeCloneError` class or raise something more generic.

Comment (jaywcarman) [reply app/models/manageiq/providers/ibm_cloud/power_virtual_servers/storage_manager/cloud_volume.rb:195]: I think this is where you would add it:
https://github.com/ManageIQ/manageiq-gems-pending/blob/master/lib/gems/pending/util/miq-exception.rb#L119

File: app/models/manageiq/providers/ibm_cloud/power_virtual_servers/storage_manager/cloud_volume.rb
Comment (jaywcarman) [inline app/models/manageiq/providers/ibm_cloud/power_virtual_servers/storage_manager/cloud_volume.rb:118]: Minor, but through this interface we can only clone a single volume so it would be clearer to keep the help text singular. (in my earlier review I lazily copy/pasted the text from the [Power Cloud API docs](https://cloud.ibm.com/apidocs/power-cloud#pcloud-v2-volumes-clone-post)).
```suggestion
          :helperText => _("Base name of the new cloned volume. The cloned Volume name will be prefixed with 'clone-' and suffixed with '-#####' (where ##### is a 5 digit random number)"),
```

**Review by jaywcarman — APPROVED** [review]
LGTM! I'm good to merge after the remaining core PRs are merged:
- https://github.com/ManageIQ/manageiq-api/pull/1164
- https://github.com/ManageIQ/manageiq-ui-classic/pull/8301
- https://github.com/ManageIQ/manageiq-gems-pending/pull/545

**Review by agrare — APPROVED** [review]

**Pull request merged**
